internal/xds/server/kubejwt/tokenreview.go

```diff
@@ -56,17 +56,18 @@ func (i *JWTAuthInterceptor) validateKubeJWT(ctx context.Context, token string)
 		return fmt.Errorf("token is not authenticated")
 	}
 
-	// TODO: (cnvergence) define a better way to check if the token is coming from the correct node
 	if tokenReview.Status.User.Extra != nil {
 		podName := tokenReview.Status.User.Extra[authPodNameKey]
 		if podName[0] == "" {
 			return fmt.Errorf("pod name not found in token review response")
 		}
-		parts := strings.Split(podName[0], "-")
-		irKey := fmt.Sprintf("%s/%s", parts[1], parts[2])
 
-		if !i.cache.SnapshotHasIrKey(irKey) {
-			return fmt.Errorf("pod %s not found in cache", podName)
+		if podName[0] != proxyMetadata.nodeID {
```

Comment:
these can also be spoofed, 
instead in the interceptor, Upton successful verification can be add 
ctx := context.WithValue(r.Context(), "envoy-namespace", namespaceInClaim)

and perform a GET later inside https://github.com/envoyproxy/gateway/blob/7afa94ae3c25f3a006c7fb678b11f22cee1f85d8/internal/xds/cache/snapshotcache.go#L177 and validate with `Node.Cluster`  https://github.com/envoyproxy/gateway/blob/7afa94ae3c25f3a006c7fb678b11f22cee1f85d8/internal/infrastructure/common/proxy_args.go#L54 or some other node metadata 
https://github.com/envoyproxy/envoy/blob/9cd454d18227dd9b7b13d5b3051304e4fdd219fb/api/envoy/api/v2/core/base.proto#L132